js/main.js

```diff
@@ -14,26 +14,46 @@ var $entryList = document.querySelector('ul');
 
 $form.addEventListener('submit', function (event) {
   event.preventDefault();
-  var newEntry = {
-    entryId: data.nextEntryId,
-    note: $form.elements.notes.value,
-    link: $form.elements.photourl.value,
-    title: $form.elements.title.value
-  };
-  data.nextEntryId++;
-  data.entries.unshift(newEntry);
-  document.querySelector('#default-pic').src = 'images/placeholder-image-square.jpg';
-  $form.reset();
-  toggleNoEntries();
-  viewSwap('entries');
-  var newLiReturn = renderEntry(newEntry);
-  $entryList.appendChild(newLiReturn);
+
+  if (data.editing === null) {
+    var newEntry = {
+      entryId: data.nextEntryId,
+      note: $form.elements.notes.value,
+      link: $form.elements.photourl.value,
+      title: $form.elements.title.value
+    };
+    data.entries.unshift(newEntry);
+    document.querySelector('#default-pic').src = 'images/placeholder-image-square.jpg';
+    $form.reset();
+    toggleNoEntries();
+    viewSwap('entries');
+    var newLiReturn = renderEntry(newEntry);
+    $entryList.appendChild(newLiReturn);
+    data.nextEntryId++;
+  } else {
+    var editingIndex = data.entries.entryId;
+    var editedEntry = {
+      entryId: editingIndex,
+      note: $form.elements.notes.value,
+      link: $form.elements.photourl.value,
+      title: $form.elements.title.value
+    };
+
+    data.entries.splice(editingIndex);
+    data.entries.unshift(editedEntry);
+    toggleNoEntries();
+    viewSwap('entries');
+    var editedLiReturn = renderEntry(editedEntry);
+    $entryList.appendChild(editedLiReturn);
```

Comment:
Same here, you shouldn't be adding it to the end of the list, you should be replacing the original DOM tree at its original position before editing.